plugin/trino-ranger/src/main/java/io/trino/plugin/ranger/RangerSystemAccessControl.java

```diff
@@ -0,0 +1,1966 @@
+
+            AccessDeniedException.denyCreateView(view.getSchemaTableName().getTableName());
+        }
+
+        LOG.debug("<== checkCanCreateView(context={}, view={}): allowed", context, view);
+    }
+
+    /**
+     * This is evaluated against the table name as ownership information is not available
+     */
+    @Override
+    public void checkCanDropView(SystemSecurityContext context, CatalogSchemaTableName view)
+    {
+        LOG.debug("==> checkCanDropView(context={}, view={})", context, view);
+
+        if (!hasPermission(createResource(view), context, TrinoAccessType.DROP, "DropView")) {
+            LOG.debug("<== checkCanDropView(context={}, view={}): denied", context, view);
+
+            AccessDeniedException.denyDropView(view.getSchemaTableName().getTableName());
+        }
+
+        LOG.debug("<== checkCanDropView(context={}, view={}): allowed", context, view);
+    }
+
+    /**
+     * This is evaluated against the table name as ownership information is not available
+     */
+    @Override
+    public void checkCanRenameView(SystemSecurityContext context, CatalogSchemaTableName view, CatalogSchemaTableName newView)
+    {
+        LOG.debug("==> checkCanRenameView(context={}, view={}, newView={})", context, view, newView);
+
+        if (!hasPermission(createResource(view), context, TrinoAccessType.ALTER, "RenameView")) {
+            LOG.debug("<== checkCanRenameView(context={}, view={}, newView={}): denied", context, view, newView);
+
+            AccessDeniedException.denyRenameView(view.toString(), newView.toString());
+        }
+
+        LOG.debug("<== checkCanRenameView(context={}, view={}, newView={}): allowed", context, view, newView);
+    }
+
+    @Override
+    public void checkCanSetViewAuthorization(SystemSecurityContext context, CatalogSchemaTableName view, TrinoPrincipal principal)
+    {
+        LOG.debug("==> checkCanSetViewAuthorization(context={}, view={}, principal={})", context, view, principal);
+
+        if (!hasPermission(createResource(view), context, TrinoAccessType.ALTER, "SetViewAuthorization")) {
+            LOG.debug("<== checkCanSetViewAuthorization(context={}, view={}, principal={}): denied", context, view, principal);
+
+            AccessDeniedException.denySetViewAuthorization(view.toString(), principal);
+        }
+
+        LOG.debug("<== checkCanSetViewAuthorization(context={}, view={}, principal={}): allowed", context, view, principal);
+    }
+
+    /**
+     * This check equals the check for checkCanCreateView
+     */
+    @Override
+    public void checkCanCreateViewWithSelectFromColumns(SystemSecurityContext context, CatalogSchemaTableName table, Set<String> columns)
+    {
+        LOG.debug("==> checkCanCreateViewWithSelectFromColumns(context={}, table={}, columns={})", context, table, columns);
+
+        try {
+            checkCanCreateView(context, table);
+        }
+        catch (AccessDeniedException ade) {
```

Comment:
```suggestion
        try {
            checkCanSelectFromColumns(context, table, columns);
        }
        catch (AccessDeniedException ade) {
```
I guess it should use `checkCanSelectFromColumns()` instead, because `checkCanCreateView()` is used in `CreateViewTask`, whereas the `checkCanCreateViewWithSelectFromColumns()` method is used in contexts like ViewAccessControl to verify that SELECT permissions in the view definition are valid.

Reply:
@okayhooni  - thank you for your review. To authorize `create view v1 as select * from t1;`, I found that Trino calls following methods (in addition to several other calls: `canAccessCatalog()`, `getRowFilters()`, `getColumnMask()`):

1. `checkCanCreateView()`
2. `checkCanSelectFromColumns()`

Do you know when `checkCanSelectFromColumns()` is used? Perhaps the expectation is to authorize both 'createView' and 'selectFromColumns' in this call?

Reply:
As you know, `checkCanSelectFromColumns()` is called every time a user attempts to select a specific column from a specific table (for example, when a user executes a query like `SELECT field_a FROM table_b`).

Since `checkCanCreateView()` is already called when a user attempts to create a view, it seems sufficient to only check `checkCanSelectFromColumns()` in this `checkCanCreateViewWithSelectFromColumns()` call